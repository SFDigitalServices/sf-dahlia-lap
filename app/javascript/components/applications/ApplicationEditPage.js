import React from 'react'

import mapProps from 'utils/mapProps'

import CardLayout from '../layouts/CardLayout'
import { saveApplication } from './actions'
import PaperApplicationForm from './application_form/PaperApplicationForm'

const ApplicationEditPageForm = ({ listing, application, editPage, lendingInstitutions }) => {
  const saveEditApplication = async (submitType, submittedValues, listing, editPage) => {
    submittedValues.listing_id = listing.id
    return saveApplication(submitType, submittedValues, listing, editPage)
  }
  return (
    <PaperApplicationForm
      listing={listing}
      application={application}
      editPage={editPage}
      onSubmit={saveEditApplication}
      lendingInstitutions={lendingInstitutions}
    />
  )
}

const ApplicationEditPage = ({ listing, application, editPage, lendingInstitutions }) => {
  const pageHeader = {
    title: 'Edit Application',
    content: `Application lottery number: ${application.lottery_number}. For listing: ${listing.name}`
  }

  return (
    <CardLayout pageHeader={pageHeader}>
      <ApplicationEditPageForm
        listing={listing}
        application={application}
        editPage={editPage}
        lendingInstitutions={lendingInstitutions}
      />
    </CardLayout>
  )
}

const mapProperties = ({ listing, application, editPage, lendingInstitutions }) => {
  return {
    listing,
    application: application,
    lendingInstitutions: lendingInstitutions,
    editPage
  }
}

export default mapProps(mapProperties)(ApplicationEditPage)
