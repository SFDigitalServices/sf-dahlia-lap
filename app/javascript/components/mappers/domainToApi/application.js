import { toPairs, filter, map, join, compact } from 'lodash'
import { createFieldMapper } from '~/utils/objectUtils'
const checkListToString = (list) => {
  return join(compact(list), ';')
}

const adaPrioritiesMapValues = {
  'vision_impairments': 'Vision impairments',
  'mobility_impairments': 'Mobility impairments',
  'hearing_impairments': 'Hearing impairments'
}

const mapAdaPrioritiesMap = (list) => {
  const selectedPriorities = filter(toPairs(list), ([a, b]) => b)
  return map(selectedPriorities, ([a, b]) => adaPrioritiesMapValues[a])
}

export const applicationFieldMapper = {
  id: 'id',
  primary_applicant_contact: 'primaryApplicantContact',
  application_language: 'applicationLanguage',
  has_military_service: 'hasMilitaryService',
  has_developmental_disability: 'hasDevelopmentalDisability',
  answered_community_screening: 'answeredCommunityScreening',
  housing_voucher_or_subsidy: 'householdVouchersSubsidies',
  terms_acknowledged: 'agreeToTerms',
  number_of_dependents: 'numberOfDependents',
  number_of_seniors: 'numberOfSeniors',
  number_of_minors: 'numberOfMinors',
  reserved_senior: 'hasSenior',
  application_submission_type: 'applicationSubmissionType',
  processing_status: 'processingStatus',
  household_assets: 'householdAssets',
  annual_income: 'annualIncome',
  monthly_income: 'monthlyIncome',
  confirmed_household_annual_income: 'confirmedHouseholdAnnualIncome',
  hh_total_income_with_assets_annual: 'HHTotalIncomeWithAssets',
  total_monthly_rent: 'totalMonthlyRent',
  adaPrioritiesSelected: (source) => checkListToString(mapAdaPrioritiesMap(source.has_ada_priorities_selected)),
  is_first_time_homebuyer: 'isFirstTimeHomebuyer',
  has_completed_homebuyer_education: 'hasCompletedHomebuyerEducation',
  has_loan_preapproval: 'hasLoanPreapproval',
  lending_agent: 'lendingAgent'
}

export const mapApplication = createFieldMapper(applicationFieldMapper)
