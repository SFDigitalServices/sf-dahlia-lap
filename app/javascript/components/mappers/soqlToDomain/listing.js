import { mapList, mapShape, compactShape } from '../utils'
import { mapOpenHouse } from './open_house'
import { mapUnit } from './unit'
import { mapListingLotteryPreference } from './listing_lottery_preference'
import { mapUser } from './user'
import { mapAccount } from './account'
import { mapBuilding } from './building'

export const mapListing = (l) => {
  return compactShape({
    owner: mapShape(mapUser, l.Owner),
    account: mapShape(mapAccount, l.Account),
    open_houses: mapList(mapOpenHouse, l.Open_Houses),
    listing_lottery_preferences: mapList(mapListingLotteryPreference, l.Listing_Lottery_Preferences),
    units: mapList(mapUnit, l.Units),
    building: mapShape(mapBuilding, l.Building),
    id: l.Id,
    owner_id: l.OwnerId,
    application_due_date: l.Application_Due_Date,
    name: l.Name,
    status: l.Status,
    min_br: l.Min_BR,
    lotterry_winners: l.Lottery_Winners,
    max_br: l.Max_BR,
    lottery_results: l.Lottery_Results,
    min_income: l.Min_Income,
    max_income: l.Max_Income,
    min_occupancy: l.Min_Occupancy,
    max_occupancy: l.Max_Occupancy,
    building_name: l.Building_Name,
    neighborhood: l.Neighborhood,
    building_street_address: l.Building_Street_Address,
    developer: l.Developer,
    building_city: l.Building_City,
    building_url: l.Building_URL,
    building_state: l.Building_State,
    year_built: l.Year_Built,
    building_zip_code: l.Building_Zip_Code,
    description: l.Description,
    lottery_prefrences: l.Lottery_Preferences,
    acessibility: l.Accessibility,
    fee: l.Fee,
    amenities: l.Amenities,
    deposit_min: l.Deposit_Min,
    costs_not_included: l.Costs_Not_Included,
    deposit_max: l.Deposit_Max,
    lottery_date: l.Lottery_Date,
    lottery_results_date: l.Lottery_Results_Date,
    lottery_venue: l.Lottery_Venue,
    lottery_summary: l.Lottery_Summary,
    lottery_street_address: l.Lottery_Street_Address,
    lottery_city: l.Lottery_City,
    lottery_url: l.Lottery_URL,
    reserved_community_type: l.Reserved_community_type,
    application_phone: l.Application_Phone,
    application_organization: l.Application_Organization,
    application_street_address: l.Application_Street_Address,
    application_city: l.Application_City,
    download_url: l.Download_URL,
    application_state: l.Application_State,
    application_postal_code: l.Application_Postal_Code,
    in_lottery: l.In_Lottery,
    leasing_agent_name: l.Leasing_Agent_Name,
    leasing_Agent_title: l.Leasing_Agent_Title,
    leasing_agent_email: l.Leasing_Agent_Email,
    leasing_agent_phone: l.Leasing_Agent_Phone,
    legal_disclaimers: l.Legal_Disclaimers,
    building_selection_criteria: l.Building_Selection_Criteria,
    pet_policy: l.Pet_Policy,
    report_id: l.Report_id,
    required_documents: l.Required_Documents,
    smoking_policy: l.Smoking_Policy,
    eviction_history: l.Eviction_History,
    criminal_history: l.Criminal_History,
    credit_rating: l.Credit_Rating,
    lottery_status: l.Lottery_Status,
    office_hours: l.Office_Hours,
    information_sessions: l.Information_Sessions,
    nflagged_applications: l.nFlagged_Applications,
    units_available: l.Units_Available,
    lease_signed_application: l.Lease_Signed_Application,
    last_modified_date: l.LastModifiedDate,
    is_sale: l.isSale,
    is_rental: l.isRental
  })
}
