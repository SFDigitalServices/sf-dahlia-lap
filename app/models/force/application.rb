# frozen_string_literal: true

# TODO: Re-enable Rubocop's line length check on this file once
# we have settled on a format for the field mapping items
# rubocop:disable Metrics/LineLength
module Force
  # Represent an application object. Provide mapping between
  # Salesforce object field names, Salesforce custom API field names,
  # and LAP domain field names for applications.
  class Application < Force::ObjectBase
    # TODO: Once we add more models and more fields, consider moving the
    # field mappings into YML files or other places.
    FIELD_NAME_MAPPINGS = [
      { custom_api: 'adaPrioritiesSelected', domain: 'has_ada_priorities_selected', salesforce: 'Has_ADA_Priorities_Selected' },
      { custom_api: 'agreeToTerms', domain: 'terms_acknowledged', salesforce: 'Terms_Acknowledged' },
      { custom_api: 'annualIncome', domain: 'annual_income', salesforce: 'Annual_Income' },
      { custom_api: 'answeredCommunityScreening', domain: 'answered_community_screening', salesforce: 'Answered_Community_Screening' },
      { custom_api: 'applicationLanguage', domain: 'application_language', salesforce: 'Application_Language' },
      { custom_api: 'applicationSubmissionType', domain: 'application_submission_type', salesforce: 'Application_Submission_Type' },
      { custom_api: 'applicationSubmittedDate', domain: 'application_submitted_date', salesforce: 'Application_Submitted_Date' },
      { custom_api: 'appRTType', domain: '', salesforce: '' },
      { custom_api: '', domain: 'createdby', salesforce: 'CreatedBy' },
      { custom_api: 'didApplicantUseHousingCounselingAgency', domain: '', salesforce: 'Applicant_used_housing_counseling_agency' },
      { custom_api: 'externalSessionId', domain: '', salesforce: 'Third_Party_External_ID' },
      { custom_api: 'finalHouseholdIncome', domain: '', salesforce: 'Final_Household_Income' },
      { custom_api: 'formMetadata', domain: '', salesforce: '' },
      { custom_api: '', domain: 'general_lottery', salesforce: 'General_Lottery' },
      { custom_api: '', domain: 'general_lottery_rank', salesforce: 'General_Lottery_Rank' },
      { custom_api: 'hasDevelopmentalDisability', domain: 'has_developmental_disability', salesforce: 'Has_DevelopmentalDisability' },
      { custom_api: 'hasMilitaryService', domain: 'has_military_service', salesforce: 'Has_Military_Service' },
      { custom_api: 'hasPublicHousing', domain: '', salesforce: '' },
      { custom_api: 'hasSenior', domain: 'reserved_senior', salesforce: 'Reserved_Senior' },
      { custom_api: 'HHTotalIncomeWithAssets', domain: 'hh_total_income_with_assets_annual', salesforce: 'HH_Total_Income_with_Assets_Annual' },
      { custom_api: 'householdAssets', domain: 'household_assets', salesforce: 'Household_Assets' },
      { custom_api: 'householdVouchersSubsidies', domain: 'housing_voucher_or_subsidy', salesforce: 'Housing_Voucher_or_Subsidy' },
      { custom_api: 'housingCounselingAgency', domain: '', salesforce: 'Which_One' },
      { custom_api: 'id', domain: 'id', salesforce: 'Id' },
      { custom_api: 'interviewScheduledDate', domain: '', salesforce: 'Interview_Scheduled_Date' },
      { custom_api: '', domain: 'is_lottery_complete', salesforce: 'Is_Lottery_Complete' },
      { custom_api: 'listingID', domain: '', salesforce: 'Listing' },
      { custom_api: 'lotteryNumber', domain: 'lottery_number', salesforce: 'Lottery_Number' },
      { custom_api: 'lotteryNumberManual', domain: 'lottery_number_manual', salesforce: 'Lottery_Number_Manual' },
      { custom_api: 'monthlyIncome', domain: 'monthly_income', salesforce: 'Monthly_Income' },
      { custom_api: 'name', domain: 'name', salesforce: 'Name' },
      { custom_api: 'numberOfDependents', domain: 'number_of_dependents', salesforce: 'Number_of_Dependents' },
      { custom_api: 'numberOfSeniors', domain: 'number_of_seniors', salesforce: 'Number_of_Seniors' },
      { custom_api: 'numberOfMinors', domain: 'number_of_minors', salesforce: 'Number_of_Minors' },
      { custom_api: 'otherHousingCounselingAgency', domain: '', salesforce: 'Other' },
      { custom_api: 'primaryApplicantContact', domain: 'primary_applicant_contact', salesforce: 'Primary_Applicant' },
      { custom_api: 'processingStatus', domain: 'processing_status', salesforce: 'Processing_Status' },
      { custom_api: 'referral', domain: 'referral_source', salesforce: 'Referral_Source' },
      { custom_api: 'snapshotId', domain: '', salesforce: 'Snapshot_ID' },
      { custom_api: 'status', domain: 'status', salesforce: 'Status' },
      { custom_api: '', domain: 'status_last_updated', salesforce: 'Status_Last_Updated' },
      { custom_api: 'subStatus', domain: 'sub_status', salesforce: 'Sub_Status' },
      { custom_api: '', domain: 'total_household_size', salesforce: 'Total_Household_Size' },
      { custom_api: 'totalMonthlyRent', domain: 'total_monthly_rent', salesforce: 'Total_Monthly_Rent' },
    ].freeze

    def to_domain
      domain_fields = super

      # Special field conversion cases for applications
      existing_fields = @fields[:salesforce].presence || @fields[:custom_api].presence
      if existing_fields.present?
        # ADA priorities
        ada_value = existing_fields['Has_ADA_Priorities_Selected'] || existing_fields['adaPrioritiesSelected']
        if ada_value
          ada_arr = ada_value.split(';')
          ada_hash = ada_arr.map { |x| [x.parameterize(separator: '_'), true] }.to_h
          domain_fields['has_ada_priorities_selected'] = ada_hash
        end

        # Created by
        domain_fields['createdby'] = { name: existing_fields['CreatedBy']['Name'] } if existing_fields['CreatedBy']
      end

      domain_fields
    end

    # TODO: There is code somewhere in this app that causes the Salesforce field
    # names we use in SOQL queries to have the "__c" suffix automatically added
    # onto them. That forces the field mappings in these Force objects to use
    # Salesforce field names without that suffix. That in turn causes issues
    # when trying to use the field mappings to translate to Salesforce fields
    # for REST API calls. Find and remove the code that strips out the "__c"
    # suffix, so that the Force objects' field mappings can contain true
    # Salesforce field names. Then remove this function as it won't be needed.
    def to_salesforce
      salesforce_fields = super
      add_salesforce_suffix(salesforce_fields)
    end
  end
end
# rubocop:enable Metrics/LineLength
