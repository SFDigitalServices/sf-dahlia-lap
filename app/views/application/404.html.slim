h1 Record not found
