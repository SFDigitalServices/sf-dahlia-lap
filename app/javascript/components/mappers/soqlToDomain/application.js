import { split, snakeCase, fromPairs } from 'lodash'
import { mapApplicationMember } from './application_member'
import { mapApplicationPreference } from './application_preference'
import { mapAttachment } from './attachment'
import { mapDemographics } from './demographics'
import { mapFlaggedApplication } from './flagged_application'
import { mapListing } from './listing'
import { mapUser } from './user'
import { mapShape, mapList } from '../utils'
import formUtils from '~/utils/formUtils'

const parseList = text => split(text, ';')
const toChecklist = list => fromPairs(list.map(i => [snakeCase(i), true]))

export const mapApplication = (a) => {
  return {
    applicant: mapShape(mapApplicationMember, a.Applicant),
    demographics: mapShape(mapDemographics, a.Applicant),
    primary_applicant_contact: a.Primary_Applicant,
    alternate_contact: mapShape(mapApplicationMember, a.Alternate_Contact),
    listing: mapShape(mapListing, a.Listing),
    preferences: mapList(mapApplicationPreference, a.preferences),
    proof_files: mapList(mapAttachment, a.proof_files),
    household_members: mapList(mapApplicationMember, a.household_members),
    flagged_applications: mapList(mapFlaggedApplication, a.flagged_applications),
    lease: a.lease,
    rental_assistances: a.rental_assistances,
    createdby: mapShape(mapUser, a.CreatedBy),
    id: a.Id,
    name: a.Name,
    status: a.Status,
    total_household_size: a.Total_Household_Size,
    application_submission_type: a.Application_Submission_Type,
    application_submitted_date: a.Application_Submitted_Date,
    annual_income: formUtils.formatPrice(a.Annual_Income),
    confirmed_household_annual_income: formUtils.formatPrice(a.Confirmed_Household_Annual_Income),
    hh_total_income_with_assets_annual: formUtils.formatPrice(a.HH_Total_Income_with_Assets_Annual),
    household_assets: formUtils.formatPrice(a.Household_Assets),
    monthly_income: a.Monthly_Income,
    housing_voucher_or_subsidy: a.Housing_Voucher_or_Subsidy,
    referral_source: a.Referral_Source,
    application_language: a.Application_Language,
    lottery_number_manual: a.Lottery_Number_Manual,
    lottery_number: a.Lottery_Number,
    total_monthly_rent: formUtils.formatPrice(a.Total_Monthly_Rent),
    general_lottery: a.General_Lottery,
    general_lottery_rank: a.General_Lottery_Rank,
    answered_community_screening: a.Answered_Community_Screening,
    has_military_service: a.Has_Military_Service,
    has_developmental_disability: a.Has_DevelopmentalDisability,
    has_ada_priorities_selected: toChecklist(parseList(a.Has_ADA_Priorities_Selected)),
    terms_acknowledged: a.Terms_Acknowledged,
    number_of_dependents: a.Number_of_Dependents,
    number_of_seniors: a.Number_of_Seniors,
    number_of_minors: a.Number_of_Minors,
    processing_status: a.Processing_Status,
    status_last_updated: a.Status_Last_Updated,
    reserved_senior: a.Reserved_Senior,
    is_first_time_homebuyer: a.Is_First_Time_Homebuyer,
    has_completed_homebuyer_education: a.Has_Completed_Homebuyer_Education,
    has_loan_preapproval: a.Has_Loan_Pre_approval,
    lending_agent: a.Lending_Agent
  }
}
