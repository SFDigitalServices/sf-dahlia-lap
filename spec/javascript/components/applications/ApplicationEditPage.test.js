/* global wait */
import React from 'react'
import renderer from 'react-test-renderer'
import { clone } from 'lodash'
import { mount } from 'enzyme'

import ApplicationEditPage from 'components/applications/ApplicationEditPage'
import listing from '../../fixtures/listing'
import application from '../../fixtures/application'
import mockApplicationApiEditPayload from '../../fixtures/application_api_edit_payload'

const mockSubmitApplication = jest.fn()

jest.mock('apiService', () => {
  return {
    submitApplication: async (data) => {
      mockSubmitApplication(data)
      return true
    }
  }
})

describe('ApplicationEditPage', () => {
  test('it should save correctly', async () => {
    const wrapper = mount(
      <ApplicationEditPage
        listing={listing}
        application={application}
        editPage />
    )

    wrapper.find('form').first().simulate('submit')

    await wait(100)

    expect(mockSubmitApplication.mock.calls.length).toBe(1)
    expect(mockSubmitApplication.mock.calls[0][0]).toEqual(mockApplicationApiEditPayload)
  })

  describe('should render', () => {
    test('successfully', () => {
      const wrapper = renderer.create(
        <ApplicationEditPage
          listing={listing}
          application={application}
          editPage />
      )

      expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('successfully with preferences', () => {
      const applicationWithPreferences = clone(application)

      applicationWithPreferences.preferences[0].Application_Member = {
        Date_of_Birth: '1981-05-04',
        First_Name: 'Flagby',
        Id: 'a0n0x000000B3xDAAS',
        Last_Name: 'Email'
      }

      expect(applicationWithPreferences.preferences).toHaveLength(1)
      expect(applicationWithPreferences.preferences[0].Application_Member).toBeTruthy()

      const wrapper = renderer.create(
        <ApplicationEditPage
          listing={listing}
          application={application}
          editPage />
      )

      expect(wrapper.toJSON()).toMatchSnapshot()
    })
  })
})
