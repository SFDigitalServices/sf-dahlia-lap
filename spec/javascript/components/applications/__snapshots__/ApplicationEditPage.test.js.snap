// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationNewPage should render succesfully 1`] = `
Array [
  <header
    className="lead-header short  bg-snow"
  >
    <div
      className="row full-width inner--3x"
    >
      <div
        className="large-12 columns"
      >
        <hgroup
          className="lead-header_group"
        >
          <h1
            className="lead-header_title small-serif c-oil"
          >
            Edit Application
          </h1>
          <div
            className="medium-8 columns no-padding"
          >
            <p>
              Application lottery number: undefined. For listing: xxx2
            </p>
          </div>
        </hgroup>
      </div>
    </div>
  </header>,
  <section
    className="tabs-card-row row full-width inner--3x margin-bottom--2x padding-top--2x"
  >
    <div
      className="app-card form-card tabs-card max-width expand-on-small"
    >
      <div>
        <form
          id="shortForm"
          onSubmit={[Function]}
        >
          <div
            className="app-card form-card medium-centered"
          >
            <div
              className="app-inner inset"
            >
              <div
                className="border-bottom margin-bottom--2x"
              >
                <div
                  className="row"
                >
                  <h3>
                    Primary Applicant
                  </h3>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="form-group"
                  >
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        First Name 
                        <span
                          className="checkbox-block_note no-margin"
                        >
                          (required)
                        </span>
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        required="true"
                        value="xxxx3"
                      />
                    </div>
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        Middle Name
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        value="middleName"
                      />
                    </div>
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        Last Name 
                        <span
                          className="checkbox-block_note no-margin"
                        >
                          (required)
                        </span>
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        required="true"
                        value="lastName"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value="xxxx2"
                    />
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Phone
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value="phone"
                    />
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Phone Type
                    </label>
                    <select
                      onBlur={[Function]}
                      onChange={[Function]}
                      value={0}
                    >
                      <option
                        disabled={true}
                        value={0}
                      >
                        Select One...
                      </option>
                      <option
                        disabled={undefined}
                        value={1}
                      >
                        Home
                      </option>
                      <option
                        disabled={undefined}
                        value={2}
                      >
                        Cell
                      </option>
                      <option
                        disabled={undefined}
                        value={3}
                      >
                        Work
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      DOB 
                      <span
                        className="checkbox-block_note no-margin"
                      >
                        - YYYY-MM-DD (required)
                      </span>
                    </label>
                    <div>
                      <div>
                        <div
                          className="react-datepicker-wrapper"
                        >
                          <div
                            className="react-datepicker__input-container"
                          >
                            <input
                              autoComplete={undefined}
                              autoFocus={undefined}
                              className=""
                              disabled={false}
                              id={undefined}
                              name={undefined}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              placeholder={undefined}
                              readOnly={undefined}
                              required={true}
                              tabIndex={undefined}
                              title={undefined}
                              type="text"
                              value="2018-04-23"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="row"
                  >
                    <h4>
                      Home Address
                    </h4>
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="form-group"
                    >
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          Street Address
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="address"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          City
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="city"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          State
                        </label>
                        <input
                          maxLength="2"
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="state"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          Zip
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="zip"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="row"
                  >
                    <h4>
                      Mailing Address
                    </h4>
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="form-group"
                    >
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          Street Address
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="mailingAddress"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          City
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="mailingCity"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          State
                        </label>
                        <input
                          maxLength="2"
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="mailingState"
                        />
                      </div>
                      <div
                        className="small-6 columns"
                      >
                        <label>
                          Zip
                        </label>
                        <input
                          onBlur={[Function]}
                          onChange={[Function]}
                          value="mailingZip"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="border-bottom margin-bottom--2x"
              >
                <div
                  className="row"
                >
                  <h3>
                    Alternate Contact
                  </h3>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="form-group"
                  >
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        First Name
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        value=""
                      />
                    </div>
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        Middle Name
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        value=""
                      />
                    </div>
                    <div
                      className="small-4 columns"
                    >
                      <label>
                        Last Name
                      </label>
                      <input
                        onBlur={[Function]}
                        onChange={[Function]}
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Alternate Contact Type
                    </label>
                    <select
                      onBlur={[Function]}
                      onChange={[Function]}
                      value={0}
                    >
                      <option
                        disabled={true}
                        value={0}
                      >
                        Select One...
                      </option>
                      <option
                        disabled={undefined}
                        value={1}
                      >
                        Family Member
                      </option>
                      <option
                        disabled={undefined}
                        value={2}
                      >
                        Friend
                      </option>
                      <option
                        disabled={undefined}
                        value={3}
                      >
                        Social Worker or Housing Counselor
                      </option>
                      <option
                        disabled={undefined}
                        value={4}
                      >
                        Other
                      </option>
                    </select>
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Alternate Contact Type Other
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    />
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Agency (if applicable)
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    />
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Phone
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    />
                  </div>
                  <div
                    className="small-4 columns"
                  >
                    <label>
                      Phone Type
                    </label>
                    <select
                      onBlur={[Function]}
                      onChange={[Function]}
                      value={0}
                    >
                      <option
                        disabled={true}
                        value={0}
                      >
                        Select One...
                      </option>
                      <option
                        disabled={undefined}
                        value={1}
                      >
                        Home
                      </option>
                      <option
                        disabled={undefined}
                        value={2}
                      >
                        Cell
                      </option>
                      <option
                        disabled={undefined}
                        value={3}
                      >
                        Work
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                className="border-bottom margin-bottom--2x"
              >
                <div
                  className="row"
                >
                  <h3>
                    Household Members
                  </h3>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="form-group"
                  >
                    <div
                      className="small-4 columns"
                    >
                      <button
                        className="mb-4 mr-4 btn btn-success"
                        onClick={[Function]}
                        type="button"
                      >
                        + Additional Member
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="border-bottom margin-bottom--2x"
              >
                <div
                  className="row"
                >
                  <div
                    className="small-12 columns"
                  >
                    <h3>
                      Reserved and Priority Qualifying Information
                    </h3>
                    <h4>
                      Qualifying Information For a Reserved Unit
                    </h4>
                  </div>
                  <div
                    className="small-12 columns margin-bottom--2x"
                  >
                    <div
                      className="small-6 columns"
                    >
                      <label>
                        ADA Priorities Selected
                      </label>
                      <div
                        className="checkbox-group"
                        role="group"
                      >
                        <div
                          className="form-item"
                        >
                          <div
                            className="checkbox"
                          >
                            <input
                              checked={false}
                              id="adaPrioritiesSelected-0"
                              name="adaPrioritiesSelected"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="checkbox"
                            />
                            <label
                              htmlFor="adaPrioritiesSelected-0"
                            >
                              Mobility impaired
                            </label>
                          </div>
                        </div>
                        <div
                          className="form-item"
                        >
                          <div
                            className="checkbox"
                          >
                            <input
                              checked={false}
                              id="adaPrioritiesSelected-1"
                              name="adaPrioritiesSelected"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="checkbox"
                            />
                            <label
                              htmlFor="adaPrioritiesSelected-1"
                            >
                              Vision impaired
                            </label>
                          </div>
                        </div>
                        <div
                          className="form-item"
                        >
                          <div
                            className="checkbox"
                          >
                            <input
                              checked={false}
                              id="adaPrioritiesSelected-2"
                              name="adaPrioritiesSelected"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="checkbox"
                            />
                            <label
                              htmlFor="adaPrioritiesSelected-2"
                            >
                              Hearing impaired
                            </label>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="border-bottom margin-bottom--2x"
              >
                <h3>
                  Preferences
                </h3>
                <div
                  className="row"
                >
                  <div
                    className="form-group"
                  >
                    <div
                      className="small-4 columns"
                    >
                      <button
                        className="mb-4 mr-4 btn btn-success"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        + Add Preference
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="border-bottom margin-bottom--2x"
              >
                <h3>
                  Declared Household Income
                </h3>
                <div
                  className="row"
                >
                  <div
                    className="small-6 columns"
                  >
                    <label>
                      Annual Income
                    </label>
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    />
                  </div>
                  <div
                    className="small-6 columns"
                  >
                    <label>
                      Housing Voucher/ Subsidy
                    </label>
                    <select
                      onBlur={[Function]}
                      onChange={[Function]}
                      value={0}
                    >
                      <option
                        disabled={true}
                        value={0}
                      >
                        Select One...
                      </option>
                      <option
                        disabled={undefined}
                        value={1}
                      >
                        true
                      </option>
                      <option
                        disabled={undefined}
                        value={2}
                      >
                        false
                      </option>
                      <option
                        disabled={undefined}
                        value={3}
                      >
                        Left Blank
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                className="checkbox-group"
                role="group"
              >
                <div
                  className="form-item"
                >
                  <div
                    className="checkbox"
                  >
                    <input
                      checked={false}
                      id="agreeToTerms"
                      name="agreeToTerms"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <label
                      htmlFor="agreeToTerms"
                    >
                      Signature on Terms of Agreement
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="button-pager"
            >
              <div
                className="button-pager_row primary"
              >
                <button
                  className="primary radius margin-right"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  Save
                </button>
                <button
                  className="primary radius"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  Save and New
                </button>
              </div>
              <div
                className="button-pager_row primary"
              >
                <a
                  className="primary radius"
                  href="/listings"
                >
                  Cancel
                </a>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </section>,
]
`;
